File: velocity-sdk-java/examples/HelloWorld.java
package io.velocity;

import java.util.Map;
import java.util.HashMap;

/**
 * Hello World example for V.E.L.O.C.I.T.Y.-WorkFlow Java SDK.
 */
public class HelloWorld {
    /**
     * Simple workflow that greets a user.
     */
    public static Object greetingWorkflow(WorkflowContext context, Object input) {
        @SuppressWarnings("unchecked")
        Map<String, Object> inputMap = (Map<String, Object>) input;
        String name = (String) inputMap.get("name");
        
        System.out.println("Workflow started: " + context.getWorkflowId());
        
        String greeting = "Hello, " + name + "! Welcome to V.E.L.O.C.I.T.Y.-WorkFlow.";
        
        System.out.println("Workflow completed: " + context.getWorkflowId());
        return greeting;
    }

    /**
     * Activity that generates a greeting.
     */
    public static Object greetActivity(ActivityContext context, Object input) {
        String name = (String) input;
        System.out.println("Activity executing: greeting " + name);
        return "Hello, " + name + "! Welcome to V.E.L.O.C.I.T.Y.-WorkFlow.";
    }

    public static void main(String[] args) throws Exception {
        // Register workflows and activities
        Map<String, java.util.function.BiFunction<WorkflowContext, Object, Object>> workflows = new HashMap<>();
        workflows.put("greeting-workflow", HelloWorld::greetingWorkflow);

        Map<String, java.util.function.BiFunction<ActivityContext, Object, Object>> activities = new HashMap<>();
        activities.put("greet-activity", HelloWorld::greetActivity);

        // Create worker
        Worker worker = new Worker(new WorkerOptions()
            .setNamespace("default")
            .setTaskQueue("greeting-queue")
            .setWorkflows(workflows)
            .setActivities(activities));

        // Start worker in a separate thread
        Thread workerThread = new Thread(() -> {
            try {
                worker.run();
            } catch (Exception e) {
                e.printStackTrace();
            }
        });
        workerThread.setDaemon(true);
        workerThread.start();

        // Give worker time to start
        Thread.sleep(1000);

        // Create client
        Client client = new Client(new ClientOptions()
            .setHostPort("localhost:7233")
            .setNamespace("default"));

        // Start workflow
        WorkflowExecution execution = client.startWorkflow(new WorkflowOptions()
            .setWorkflowId("greeting-" + System.currentTimeMillis())
            .setWorkflowType("greeting-workflow")
            .setTaskQueue("greeting-queue")
            .setInput(Map.of("name", "World")));

        System.out.println("Started workflow: " + execution.getWorkflowId());

        // Wait for result
        WorkflowHandle handle = client.getWorkflow(execution.getWorkflowId());
        Object result = handle.getResult();

        System.out.println("Workflow result: " + result);

        // Cleanup
        worker.stop();
        client.close();
    }
}

File: velocity-sdk-python/examples/hello_world.py
"""Hello World example for V.E.L.O.C.I.T.Y.-WorkFlow Python SDK"""

import time
from velocity import (
    Client,
    ClientOptions,
    Worker,
    WorkerOptions,
    WorkflowOptions,
    WorkflowContext,
    ActivityContext,
)


def greeting_workflow(context: WorkflowContext, input: dict) -> str:
    """Simple workflow that greets a user"""
    name = input["name"]
    print(f"Workflow started: {context.workflow_id}")
    
    # In a real implementation, this would execute an activity
    greeting = f"Hello, {name}! Welcome to V.E.L.O.C.I.T.Y.-WorkFlow."
    
    print(f"Workflow completed: {context.workflow_id}")
    return greeting


def greet_activity(context: ActivityContext, name: str) -> str:
    """Activity that generates a greeting"""
    print(f"Activity executing: greeting {name}")
    return f"Hello, {name}! Welcome to V.E.L.O.C.I.T.Y.-WorkFlow."


def main():
    # Create worker
    worker = Worker(WorkerOptions(
        namespace="default",
        task_queue="greeting-queue",
        workflows={
            "greeting-workflow": greeting_workflow,
        },
        activities={
            "greet-activity": greet_activity,
        },
    ))
    
    # Start worker in a separate thread
    import threading
    worker_thread = threading.Thread(target=worker.run)
    worker_thread.daemon = True
    worker_thread.start()
    
    # Give worker time to start
    time.sleep(1)
    
    # Create client
    client = Client(ClientOptions(
        host_port="localhost:7233",
        namespace="default",
    ))
    
    # Start workflow
    execution = client.start_workflow(WorkflowOptions(
        workflow_id=f"greeting-{int(time.time() * 1000)}",
        workflow_type="greeting-workflow",
        task_queue="greeting-queue",
        input={"name": "World"},
    ))
    
    print(f"Started workflow: {execution.workflow_id}")
    
    # Wait for result
    handle = client.get_workflow(execution.workflow_id)
    result = handle.result()
    
    print(f"Workflow result: {result}")
    
    # Cleanup
    worker.stop()
    client.close()


if __name__ == "__main__":
    main()
